mm2src/coins/z_coin/storage/z_change_notes/mod.rs

```diff
@@ -0,0 +1,124 @@
+    SqliteError(String),
+    #[cfg(target_arch = "wasm32")]
+    #[display(fmt = "IndexedDb Error: {_0}")]
+    #[from_stringify("InitDbError", "DbTransactionError")]
+    IndexedDbError(String),
+}
+
+#[cfg(test)]
+pub(super) mod change_note_test {
+    use crate::z_coin::storage::z_change_notes::LockedNotesStorage;
+
+    use mm2_test_helpers::for_tests::mm_ctx_with_custom_db;
+
+    const MY_ADDRESS: &str = "my_address";
+
+    pub(crate) async fn test_insert_and_remove_note_impl() {
+        let ctx = mm_ctx_with_custom_db();
+        let db = LockedNotesStorage::new(ctx, MY_ADDRESS.to_string()).await.unwrap();
+
+        // insert note
+        let result = db.insert_note("some_hex".to_string(), String::from("228")).await;
+        assert!(result.is_ok());
+
+        // remove note
+        let result = db.remove_note("some_hex".to_owned()).await;
+        assert!(result.is_ok());
+
+        // get notes(should be empty)
+        let notes = db.load_all_notes().await.unwrap();
+        assert!(notes.is_empty());
+    }
+
+    pub(crate) async fn test_load_all_notes_impl() {
+        let ctx = mm_ctx_with_custom_db();
+        let db = LockedNotesStorage::new(ctx, MY_ADDRESS.to_string()).await.unwrap();
+
+        let result = db.insert_note("some_hex".to_string(), String::from("40")).await;
+        assert!(result.is_ok());
+
+        let notes = db.load_all_notes().await.unwrap();
+        assert!(notes.len() == 1);
+    }
+
+    pub(crate) async fn test_sum_changes_impl() {
+        let ctx = mm_ctx_with_custom_db();
+        let db = LockedNotesStorage::new(ctx, MY_ADDRESS.to_string()).await.unwrap();
+
+        let result = db.insert_note("some_hex".to_string(), String::from("50")).await;
+        assert!(result.is_ok());
+
+        let result = db.insert_note("another_hex".to_string(), String::from("5")).await;
+        assert!(result.is_ok());
+    }
+}
+
+#[cfg(all(test, not(target_arch = "wasm32")))]
+mod change_note_native_tests {
+    use common::block_on;
+
+    use super::change_note_test::{test_insert_and_remove_note_impl, test_load_all_notes_impl, test_sum_changes_impl};
+
+    #[test]
+    fn test_insert_and_remove_note() { block_on(test_insert_and_remove_note_impl()) }
+
+    #[test]
+    fn test_load_all_notes() { block_on(test_load_all_notes_impl()) }
+
+    #[test]
+    fn test_sum_changes() { block_on(test_sum_changes_impl()) }
+}
+
+#[cfg(all(test, target_arch = "wasm32"))]
+mod change_note_native_tests {
+    use super::change_note_test::{test_insert_and_remove_note_impl, test_load_all_notes_impl, test_sum_changes_impl};
+    use wasm_bindgen_test::wasm_bindgen_test;
+
+    #[wasm_bindgen_test]
+    async fn test_insert_and_remove_note() { test_insert_and_remove_note_impl().await }
+
+    #[wasm_bindgen_test]
+    async fn test_load_all_notes() { test_load_all_notes_impl().await }
+
+    #[wasm_bindgen_test]
+    async fn test_sum_changes() { test_sum_changes_impl().await }
+}
```

Comment:
I wonder if this can be unified with `cross_test` macro :thinking:

Reply:
yes for sure, thanks

Reply:
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2331/commits/ddc18b5790cc451958b9bd038c755aa7835d4f40